src/main/kotlin/com/quipalup/katydid/logentry/domain/LogEntryRepository.kt

```diff
@@ -0,0 +1,9 @@
+package com.quipalup.katydid.logentry.domain
+
+import arrow.core.Either
+import com.quipalup.katydid.logentry.logentry.primaryadapter.httprest.LogEntryRequestDocument
+import com.quipalup.katydid.logentry.logentry.primaryadapter.httprest.LogEntryResponseDocument
+
+interface LogEntryRepository {
+    fun create(createRequest: LogEntryRequestDocument): Either<LogEntryError, LogEntryResponseDocument>
```

Comment:
```suggestion
    fun create(logEntry: LogEntry): Either<CreateLogEntryError, LogEntry>
```
Here we are in the most inner part of our achitecture (domain) that is made of **domain entities**, **domain services** (when needed) and **interfaces**(ports) to interact with secondary adapters that handle infrastructure communication. In this case, we will see in the future that the implementation of this interface will handle database communication.